### PR TITLE
[types] Refactor AccountAddress to move_core_types

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

This is the very first step for the libra_types refactor. The plan is that we will move all move related concept from libra_types into move_core_types, including AccountAddress, language_storage, and MoveResource. This should trim the size of libra_types crate.

I still reexport the AccountAddress in libra_types to avoid a huge change in the existing codebase.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Comment by dariorussi** [comment]
adding the rest of the Move VM team.
This is an important change so if anybody wants to chime in, this is your chance.
Dependencies for Move are tricky and `AccountAddress` is the most immediate problem we have in creating circular dependencies. 
I think shortly @runtian-zhou plans to move most of Move related common code from libra/types to language/move-core. We will see how things turn out.
I like this first step and I think we should move in this direction and revisit once move-core takes a more clear shape.

File: types/src/proptest_types.rs [hunk 1]
```diff
@@ -357,7 +357,7 @@ impl SignatureCheckedTransaction {
     ) -> impl Strategy<Value = Self> {
         (keypair_strategy, payload_strategy)
             .prop_flat_map(|(keypair, payload)| {
-                let address = AccountAddress::from_public_key(&keypair.public_key);
+                let address = from_public_key(&keypair.public_key);
```
Comment (davidiw) [inline types/src/proptest_types.rs:360]: The right way to do these imports is account_address::from_public_key and avoid importing functions directly.

File: language/move-core/types/src/unit_tests/address_test.rs [hunk 2]
```diff
@@ -52,7 +52,7 @@ fn test_address() {
     });
 
     let hash_vec =
-        &Vec::from_hex("87090718c95e76db89afa7148a6294e3a2ce7aacddcb8d21172dd8cce1a3bbd6")
+        &Vec::from_hex("c965dfb50b4ebcf000d256a4b23d36ca49f0b05a905e8124ba2f8ff3112bf662")
```
Comment (runtian-zhou) [inline language/move-core/types/src/unit_tests/address_test.rs:55]: Note the hash was modified because of the move.

**Pull request closed without merging**